merlin-server/sql/merlin/tables/activity_directive.sql

```diff
@@ -13,15 +13,21 @@ create table activity_directive (
   last_modified_arguments_at timestamptz not null default now(),
   metadata merlin_activity_directive_metadata_set default '{}'::jsonb,
 
+  anchor_ID integer default null,
+  anchored_to_start boolean default true not null,
   constraint activity_directive_natural_key
     primary key (id, plan_id),
   constraint activity_directive_owned_by_plan
     foreign key (plan_id)
     references plan
     on update cascade
     on delete cascade,
-  constraint activity_directive_start_offset_is_nonnegative
-    check (start_offset >= '0')
+  -- An activity cannot anchor to an activity in another plan
```

Comment:
I agree with this comment (I had not considered the possibility to anchor outside of the plan - I do agree that that kind of integrity should be enforced).

I'm having trouble seeing how the constraint below relates to this comment, could you help guide me through that?

Reply:
Sure! The constraint says that if we have activity A with the following (relevant) fields: `id`, `plan_id`, `anchor_id`, that there must be another activity B in `activity_directives` such that `B.id = A.anchor_id` and `B.plan_id = A.plan_id` (plus or minus the foreign key null-value allowances).

Although looking it back over, I'm not certain if I want `on delete/update cascade`.